### PR TITLE
Create translation_RO.json

### PR DESCRIPTION
**Pull request by discip**
as requested by @cipyyy here https://github.com/Ralim/IronOS/issues/1026

@sandmanRO
Could you please have a look at this and translate the bits, which were not translated yet?

### CODE REVIEW

**Comment by Ralim** [comment]
The current OLED font does not have the symbols `{'„', 'Î', '”', 'ş', 'ț', 'â', 'Ă', 'ș', 'ă', 'î'}` defined

**Comment by discip** [comment]
Thank you for the spontaneous reaction!  😃👍

1. Where does this 'result' pop up, what am I missing?
I do not want to bother you every time, so I would like to be able to find it myself. 😁
2. Is there a way to fix this, or do I have to remove those symbols?

**Comment by sandmanRO** [comment]
Oh, it is Romanian. It needs Latin extended in order to work.

**Comment by sandmanRO** [comment]
I believe you also recently added some new entries not included with this translation.

**Comment by discip** [comment]
@sandmanRO
Yes, I thought about that too, and changed that bit, but without any luck.
If I got @Ralim right, there are no such symbols in any defined tables used here yet.

**Comment by Ralim** [comment]
Ah you can see it by clicking on the red X->details then going to the logs of one of the builds.
Then in the logs you can see `KeyError: "Symbols not found in specified fonts: {'ț', '”', 'ș', '„'}"`
Which is its way of saying that the letters `ț` `”` `ș` `„` dont exist in [here](https://github.com/Ralim/IronOS/blob/master/Translations/font_tables.py) 

If you draw them using the bitmap editor (html file in that folder) or any other tool, export the hex and add them to that file then it will work.

**Comment by sandmanRO** [comment]
I thought I saw those symbols in font_tables.py but I might be wrong.

**Comment by sandmanRO** [comment]
by the way discip, are you building under Windows?

**Comment by Ralim** [comment]
> I thought I saw those symbols in font_tables.py but I might be wrong.

Could be identical looking but different codepoint symbols. Thats happened before 🙃

**Comment by sandmanRO** [comment]
I see...the reason I was asking about Windows...under Windows the default java / python encoding is not UTF-8 but iso-8859-1

**Comment by sandmanRO** [comment]
so you have to modify some of the files you are using (I will check in a second their name) to explicitly specify the UTF-8 encoding

**Comment by discip** [comment]
> I see...the reason I was asking about Windows...under Windows the default java / python encoding is not UTF-8 but iso-8859-1

I downloaded the file provided by @cipyyy and edited it in **`notepad++`**.

**Comment by sandmanRO** [comment]
Yes, but I believe the job of adding the language to your firmware is done inside make_translation.py (my apologies if I'm wrong)

**Comment by sandmanRO** [comment]
...and that is python code whicj would try to load a UTF-8 encoded file as iso-8859-1

**Comment by discip** [comment]
@sandmanRO
Sorry, but I am not building anything on my machine, since I change the files here and the rest is done automatically. 😁

**Comment by sandmanRO** [comment]
and that would generate all sort of problems under Windows 10 for anything that is not Latin basic.

**Comment by sandmanRO** [comment]
I see...I'm not familiar with the way GitHub works

**Comment by discip** [comment]
@sandmanRO
Could you please tell me, if there are alternatives for these letters?
# ș ț

**Comment by sandmanRO** [comment]
s and t

**Comment by sandmanRO** [comment]
The use of diacritics is not critical... I personally don't use them when I write in Romanian on computer

**Comment by sandmanRO** [comment]
Thank you @Ralim. I knew nothing about GitHub until not so long ago (not that know more now)

**Comment by sandmanRO** [comment]
@discip, 100% it's ok to replace the special characters with their plain ASCII "counterpart".

**Comment by Ralim** [comment]
Github is a massive thing that has grown massively.
The Continious Integration (CI) is the checks, and they are there so that you cant merge in changes that break the build for any device on any language. (I cant actually click the merge button unless they all pass).

It does make things like this annoying, but it keeps the main codebase "working" (roughly, it cant spot code issues that still compile).


I would go for just subbing in closest characters for now and then if desired someone can loop back and add the better ones later on

File: Translations/translation_RO.json [hunk 1]
```diff
@@ -0,0 +1,214 @@
...
+  "tempUnitFahrenheit": true,
+  "messages": {
+    "SettingsCalibrationDone": "Calibration done!",
+    "SettingsCalibrationWarning": "Vă rugăm să vă asigurati că vârful este la temperatura camerei, înainte de a continua!",
+    "SettingsResetWarning": "Sigur doriti să restaurati la setările implicite?",
+    "UVLOWarningString": "DC LOW",
+    "UndervoltageString": "Sub tensiune",
+    "InputVoltageString": "Intrare V: ",
+    "WarningTipTempString": "Tip temp: ",
+    "BadTipString": "BAD TIP",
```
Comment (discip) [inline Translations/translation_RO.json:14]: @sandmanRO

File: Translations/translation_RO.json [hunk 2]
```diff
@@ -0,0 +1,214 @@
+{
+  "languageCode": "RO",
+  "languageLocalName": "Română",
+  "fonts": ["ascii_basic", "latin_extended"],
+  "tempUnitFahrenheit": true,
+  "messages": {
+    "SettingsCalibrationDone": "Calibration done!",
```
Comment (discip) [inline Translations/translation_RO.json:7]: @sandmanRO

File: Translations/translation_RO.json [hunk 3]
```diff
@@ -0,0 +1,214 @@
...
+    "SettingsResetWarning": "Sigur doriti să restaurati la setările implicite?",
+    "UVLOWarningString": "DC LOW",
+    "UndervoltageString": "Sub tensiune",
+    "InputVoltageString": "Intrare V: ",
+    "WarningTipTempString": "Tip temp: ",
+    "BadTipString": "BAD TIP",
+    "SleepingSimpleString": "Zzzz",
+    "SleepingAdvancedString": "Adormit...",
+    "WarningSimpleString": "HOT!",
+    "WarningAdvancedString": "!!! HOT TIP !!!",
```
Comment (discip) [inline Translations/translation_RO.json:18]: @sandmanRO

File: Translations/translation_RO.json (hunk 1 above)
Comment (discip) [inline Translations/translation_RO.json:14]: @sandmanRO

File: Translations/translation_RO.json [hunk 4]
```diff
@@ -0,0 +1,214 @@
...
+    "SleepingTipAdvancedString": "Tip:",
+    "IdleTipString": "Tip:",
+    "IdleSetString": " Set:",
+    "TipDisconnectedString": "FĂRĂ SFAT",
+    "SolderingAdvancedPowerPrompt": "Putere: ",
+    "OffString": "Off",
+    "YourGainMessage": "Your gain:"
```
Comment (discip) [inline Translations/translation_RO.json:25]: @sandmanRO

File: Translations/translation_RO.json [hunk 5]
```diff
@@ -0,0 +1,214 @@
...
+  },
+  "messagesWarn": {
+    "ResetOKMessage": "Reset OK",
+    "SettingsResetMessage": ["Setările au fost", "resetate!"],
+    "NoAccelerometerMessage": ["Nu accelerometru", "detectat!"],
+    "NoPowerDeliveryMessage": ["Fără USB-PD IC", "detectat!"],
+    "LockingKeysString": "BLOCAT",
+    "UnlockingKeysString": "DEBLOCAT",
+    "WarningKeysLockedString": "!BLOCAT!",
+    "WarningThermalRunaway": ["Thermal", "Runaway"]
```
Comment (discip) [inline Translations/translation_RO.json:35]: @sandmanRO

File: Translations/translation_RO.json [hunk 6]
```diff
@@ -0,0 +1,214 @@
...
+      "text2": ["Viteză", "derulare"],
+      "desc": "Viteză derulare text cu informatii la (S=lent | F=rapid)"
+    },
+    "QCMaxVoltage": {
+      "text2": ["QC", "voltaj"],
+      "desc": "Tensiunea maximă QC dorită pentru care negociază letconul"
+    },
+    "PDNegTimeout": {
+      "text2": ["PD", "timeout"],
+      "desc": "PD negotiation timeout in 100ms steps for compatibility with some QC chargers"
```
Comment (discip) [inline Translations/translation_RO.json:151]: @sandmanRO

File: Translations/translation_RO.json [hunk 7]
```diff
@@ -0,0 +1,214 @@
...
+      "text2": ["Întârziere", "puls putere"],
+      "desc": "Întârziere înainte de declansarea pulsului trezit (x 2.5s)"
+    },
+    "PowerPulseDuration": {
+      "text2": ["Durată", "puls putere"],
+      "desc": "Păstrati-pulsul-treaz durată (x 250ms)"
+    },
+    "LanguageSwitch": {
+      "text2": ["Limbă:", " RO     Română"],
+      "desc": "Current firmware language"
```
Comment (discip) [inline Translations/translation_RO.json:203]: @sandmanRO

File: Translations/translation_RO.json [hunk 8]
```diff
@@ -0,0 +1,214 @@
...
+    },
+    "Brightness": {
+        "text2": ["Screen", "contrast"],
+        "desc": "Adjust the brightness of the OLED screen"
```
Comment (discip) [inline Translations/translation_RO.json:207]: @sandmanRO

File: Translations/translation_RO.json [hunk 9]
```diff
@@ -0,0 +1,214 @@
...
+    },
+    "ColourInversion": {
+        "text2": ["Invert", "screen"],
+        "desc": "Invert the colours of the OLED screen"
```
Comment (discip) [inline Translations/translation_RO.json:211]: @sandmanRO

Comment (sandmanRO) [reply Translations/translation_RO.json:211]: "Inverseaza culoarea ecranului" or "Schema de culoare negativa"

File: Translations/translation_RO.json (hunk 7 above)
Comment (sandmanRO) [reply Translations/translation_RO.json:203]: Limba curenta

File: Translations/translation_RO.json (hunk 6 above)
Comment (sandmanRO) [reply Translations/translation_RO.json:151]: Timp limita de negociere pentru tranzactia PD, in pasi de 100ms, pentru compatibilitate cu alimentatoarele QC

File: Translations/translation_RO.json (hunk 7 above)
Comment (discip) [reply Translations/translation_RO.json:203]: No problem.
As long as you like it, I will write it down for you.

File: Translations/translation_RO.json (hunk 5 above)
Comment (sandmanRO) [reply Translations/translation_RO.json:35]: I would translate this by meaning rather by words, something like: "Incalzire" "Esuata" (it means failed heating)

File: Translations/translation_RO.json (hunk 4 above)
Comment (sandmanRO) [reply Translations/translation_RO.json:25]: "TipDisconnectedString": "FARA VARF" or "VARF LIPSA"
"OffString": "Nu",
"YourGainMessage": "Coeficient:"
(power is ok)

File: Translations/translation_RO.json (hunk 1 above)
Comment (sandmanRO) [reply Translations/translation_RO.json:14]: "WarningTipTempString": "Temp. varf: ",
"BadTipString": "VARF DEFECT",

**Comment by sandmanRO** [comment]
line 210: "text2": ["Inverseaza", "culoare"],
line 211. "desc": "Inverseaza culoarea ecranului" 

Sorry for my slow reaction (in my defence, here is about 4:30AM)

**Comment by discip** [comment]
> line 210: "text2": ["Inverseaza", "culoarea"],
> line 211. "desc": "Inverseaza culoarea ecranului"
> 
> Sorry for my slow reaction (in my defence, here is about 4:30AM)

No worries, did already change it.
Thank you. 👍

**Comment by discip** [comment]
@sandmanRO
Please have a final look over this once more, before asking @Ralim to merge this in. 😊

thanks in advance

**Comment by Ralim** [comment]
I'll wait for Sandman to wake up and confirm this looks good, then will merge this in :)

**Comment by cipyyy** [comment]
Hello,
Thank you very for the interest and for the effort made to correct the translation.
Good luck!

**Comment by sandmanRO** [comment]
Hello Gentlemen, I would do few small changes:

Line 171:  "desc": "Intensitatea puterii la păstrati-pulsul-treaz (watt)"
I would replace with:
"desc": "Intensitatea puterii pulsului de mentinere activa a blocului de alimentare (wat)"

Line 195: "desc": "Întârziere înainte de declansarea pulsului trezit (x 2.5s)"
with
"desc": "Perioada pulsului de mentinere (x 2.5s)"

Line 199: "desc": "Păstrati-pulsul-treaz durată (x 250ms)"
with
"desc": "Durata pulsului de mentinere (x 250ms)"

There is no "good" translation for "keep-awake". Obviously you can translate each word, as originally done, but it would not have too much sense (as technical term)

**Comment by discip** [comment]
Will change this later.

**Comment by discip** [comment]
@Ralim
Should be good to go now. 😊

**Comment by cipyyy** [comment]
It is very good.
Thanks!

**Comment by discip** [comment]
@sandmanRO
## Thank you for sorting this out! 😃👍

**Pull request merged**